src/OgMembershipAccessControlHandler.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Drupal\og;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Entity\EntityHandlerInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines the OG membershipimplementation for entity access control handler.
+ */
+class OgMembershipAccessControlHandler extends EntityAccessControlHandler implements EntityHandlerInterface {
+
+  /**
+   * The OG access service.
+   *
+   * @var \Drupal\og\OgAccess
+   */
+  protected $ogAccess;
+
+  /**
+   * Constructs a OgMembershipAccessControllHandler object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type definition.
+   * @param \Drupal\og\OgAccessInterface $og_access
+   *   The OG access service.
+   */
+  public function __construct(EntityTypeInterface $entity_type, OgAccessInterface $og_access) {
+    $this->entityTypeId = $entity_type->id();
+    $this->entityType = $entity_type;
+    $this->ogAccess = $og_access;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
+    return new static(
+      $entity_type,
+      $container->get('og.access')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    $group = $entity->getGroup();
+
+    // Do not allow deleting the group owner's membership.
+    if (($operation === 'delete') && ($group instanceof EntityOwnerInterface) && ($group->getOwnerId() == $entity->getOwner()->id())) {
+      return AccessResult::forbidden();
+    }
+
+    // If the user has permission to administer all groups, allow access.
+    if ($account->hasPermission('administer organic groups')) {
+      return AccessResult::allowed();
+    }
+
+    $permissions = [OgAccess::ADMINISTER_GROUP_PERMISSION, 'manager members'];
```

Comment:
I see that here and in 1 other place in the PR the permission `manager members` is used, but in the rest of the PR I see `manage members`. I assume these should all be `manage members`.

Reply:
Indeed
```suggestion
    $permissions = [OgAccess::ADMINISTER_GROUP_PERMISSION, 'manage members'];
```